packages/vm/src/traits.rs

```diff
@@ -27,7 +27,26 @@ impl<S: Storage, A: Api, Q: Querier> Extern<S, A, Q> {
 }
 
 #[cfg(feature = "iterator")]
-pub type StorageIteratorItem = FfiResult<(KV, u64)>;
+pub type NextItem = (Option<KV>, u64);
+
+#[cfg(feature = "iterator")]
+pub trait StorageIterator {
+    fn next(&mut self) -> FfiResult<NextItem>;
+
+    /// Collects all elements, ignoring gas costs
+    fn elements(&mut self) -> FfiResult<Vec<KV>> {
```

Comment:
I tried

```rust
fn elements(self) -> FfiResult<Vec<KV>> {
where
        Self: Sized,
```

here like it is done in https://doc.rust-lang.org/std/iter/trait.Iterator.html#method.count or https://doc.rust-lang.org/std/iter/trait.Iterator.html#method.collect but that gave me all kinds of compile errors.

Reply:
A bunch of those

```
error: the `elements` method cannot be invoked on a trait object
   --> packages/vm/src/testing/storage.rs:164:33
    |
164 |             let elements = iter.elements().unwrap();
    |                                 ^^^^^^^^
    | 
   ::: packages/vm/src/traits.rs:39:15
    |
39  |         Self: Sized,
    |               ----- this has a `Sized` requirement
```

Reply:
addressed in 3a93ab8